lib/LANraragi/Plugin/Metadata/RegexParse.pm

```diff
@@ -64,42 +130,87 @@ sub get_tags {
     if ( defined $5 ) { $title    = trim($5); }
     if ( defined $7 ) { $series   = $7; }
     if ( defined $9 ) { $language = $9; }
+    my $tail = trim( $+{'tail'} );
+
+    if ($tail) {
+
+        # match trailing_tags (...{Tags}.ext)
+        if ( $params->{'check_trailing_tags'} ) {
+            $tail =~ /(?<head>.*)\{(?<ttags>[^\}]*)\}$/;
+            $trailing_tags = $+{'ttags'};
+            $tail          = $+{'head'};
+        }
 
-    my @tags = ();
+        # match any remaining parenthesis
+        if ( $tail && $params->{'keep_all_captures'} ) {
+            my @items = ( $tail =~ /\(([^\)]+)\)|\{([^}]+)\}|\[([^\]]+)\]/g );
+            $other_captures = join( ',', grep { trim($_) } @items );
+        }
+    }
+
+    my @tags;
 
-    if ( $event ne "" ) {
-        push @tags, "event:$event";
+    push @tags, parse_artist_value($artist)                                           if ($artist);
+    push @tags, "event:$event"                                                        if ($event);
+    push @tags, parse_language_value($language)                                       if ($language);
+    push @tags, parse_captured_value_for_namespace( $series, 'series:' )              if ($series);
+    push @tags, parse_captured_value_for_namespace( $other_captures, $PLUGIN_TAG_NS ) if ($other_captures);
+    push @tags, parse_captured_value_for_namespace( $trailing_tags, '' )              if ($trailing_tags);
+
+    if ( !$params->{'keep_all_captures'} ) {
+        @tags = grep { !m/^\Q$PLUGIN_TAG_NS/ } @tags;
     }
 
-    if ( $artist ne "" ) {
+    return ( join( ", ", sort @tags ), trim($title) );
+}
 
-        #Special case for circle/artist sets:
-        #If the string contains parenthesis, what's inside those is the artist name
-        #the rest is the circle.
-        if ( $artist =~ /(.*) \((.*)\)/ ) {
-            push @tags, "group:$1";
-            push @tags, "artist:$2";
+sub parse_language_value {
+    my ($language) = @_;
+    my @tags;
+    my @maybe_languages = map { trim( lc $_ ) } split( m/,/, $language );
```

Comment:
This feels too complex to have in a parsing plugin - I would just scrap the VALID_LANGUAGES part entirely and encourage the use of tag rules for users that want to coalesce their language tags.

Reply:
You're right, but I thought that since they are quite easily identifiable, we could save a few lines of rules.

Reply:
On second thought, the language map is ugly to look at and replacing it with a library is excessive.
On the other hand how many languages ​​would one ever want to manage?
You've convinced me :smiley: